File: server/php/store/Shipping.php
<?php

namespace Store;

class Shipping
{
  // This would be a lookup in your database
  public static $options = [
    'free' => [ 'label' => 'Free Shipping', 'detail' => 'Delivery within 5 days', 'amount' => 0 ],
    'express' => [ 'label' => 'Express Shipping', 'detail' => 'Next day delivery', 'amount' => 500 ]
  ];

  public static function getShippingOptions() {
    $shippingOptions = [];
    foreach (self::$options as $id => $option) {
      $shippingOptions[] = array_merge([ 'id' => $id ], $option);
    }
    
    return $shippingOptions;
  }

  public static function getShippingCost($id) {
    if (isset(self::$options[$id])) {
      return self::$options[$id]['amount'];
    }

    throw new \UnexpectedValueException('Unknown shipping option ID. Argument passed: ' . $id);
  }
}

File: server/go/inventory/shipping.go
package inventory

type ShippingOption struct {
	ID     string `json:"id"`
	Label  string `json:"label"`
	Detail string `json:"detail"`
	Amount int64  `json:"amount"`
}

func ShippingOptions() []ShippingOption {
	return []ShippingOption{
		{
			ID:     "free",
			Label:  "Free Shipping",
			Detail: "Delivery within 5 days",
			Amount: 0,
		},
		{
			ID:     "express",
			Label:  "Express Shipping",
			Detail: "Next day delivery",
			Amount: 500,
		},
	}
}

func GetShippingCost(optionID string) (int64, bool) {
	for _, option := range ShippingOptions() {
		if option.ID == optionID {
			return option.Amount, true
		}
	}

	return 0, false
}
